lib/shopify_app/controller_concerns/embedded_app.rb

```diff
@@ -42,5 +44,14 @@ def set_esdk_headers
       response.set_header("P3P", 'CP="Not used"')
       response.headers.except!("X-Frame-Options")
     end
+
+    def deduced_phishing_attack?(decoded_host)
+      sanitized_host = ShopifyApp::Utils.sanitize_shop_domain(decoded_host)
+      if sanitized_host.nil?
+        ShopifyApp::Logger.info("host param from callback is not from a trusted domain")
+        ShopifyApp::Logger.info("redirecting to root as this is likely a phishing attack")
```

Comment:
I'd rather log a multi-line message here (or ideally a single line), since logs won't necessarily just be printed out in a console.